tests/scripts/components-compiler.sh

```diff
@@ -93,10 +93,7 @@ component_test_gcc15_drivers_opt () {
     scripts/config.py full
     loc_cflags="$ASAN_CFLAGS -DPSA_CRYPTO_DRIVER_TEST -DMBEDTLS_CONFIG_ADJUST_TEST_ACCELERATORS"
     loc_cflags="${loc_cflags} -I../framework/tests/include -O2"
-    # Until https://github.com/Mbed-TLS/mbedtls/issues/9814 is fixed,
-    # disable the new problematic optimization.
-    loc_cflags="${loc_cflags} -fzero-init-padding-bits=unions"
-    # Also allow a warning that we don't yet comply to.
+    # Allow a warning that we don't yet comply to.
```

Comment:
Actually, looking at current development if looks like this change was already done. I'm surprised github doesn't show it as a conflict. But overall I'm not sure we really need this PR, considering the only changes it's doing are either already done, or submodule updates.